src/core/include/iDynTree/Core/Rotation.h

```diff
@@ -271,6 +296,19 @@ namespace iDynTree
          */
         static Rotation Identity();
 
+        /**
+         * Construct a rotation matrix from the given quaternion representation
+         *
+         * The quaternion is expected to be ordered in the following way:
+         * - s \in \mathbb{R} the constituent of the quaterion
+         * - r \in \mathbb{R}^3 the imaginary part of the quaternion
+         *
+         * @param quaternion a quaternion representing a rotation
+         *
+         * @return The rotation matrix
+         */
+        static Rotation FromQuaternion(const iDynTree::Vector4& quaternion);
```

Comment:
should we find another name?
I don't like `Quaternion` as it is not clear that we are creating a rotation matrix from a quaternion.

`Rotation::RotationFromQuaternion` ?